src/librustc_typeck/check/mod.rs

```diff
@@ -4964,6 +4964,18 @@ impl<'a, 'tcx> FnCtxt<'a, 'tcx> {
         }
     }
 
+    fn is_unsized_return(&self, blk_id: hir::HirId) -> bool {
```

Comment:
Would be more informative as `fn is_return_ty_unsized`

Reply:
Also, please move this to `coercion.rs`. This file is massive and unstructured. We should avoid adding random stuff to it.